experimental/Origami/src/Origami.jl

```diff
@@ -0,0 +1,333 @@
+end
+
+function veech_group(O::Origami)
+    # TODO use hashtables or not? Implement modular subgroup?
+    GAP.Globals.ComputeVeechGroupWithHashTables(GapObj(O))
+end
+
+function index_monodromy_group(o::Origami)
+    return GAP.Globals.IndexOfMonodromyGroup(GapObj(o))::Int
+end
+
+function sum_of_lyapunov_exponents(o::Origami)
+    return QQFieldElem(GAP.Globals.SumOfLyapunovExponents(GapObj(o)))
+end
+
+function translations(o::Origami)
+    h = horizontal_perm(o)
+    v = vertical_perm(o)
+    G = normalform_conjugators(o)
+
+    O = (i -> origami(i^-1 * h * i, i^-1 * v * i)).(G)
+    list = [cperm([degree(o)])]
+
+    l = length(O)
+    for i in 1:l
+        positions = findall(item -> item == O[i], O)
+        if length(positions) != 1
+            for j in positions
+                push!(list, G[i] * G[j]^-1)
+            end
+        end
+    end
+
+    return collect(Set(list))
+end
+
+function is_hyperelliptic(o::Origami)
+    # check whether -1 is in the veech group
+	if !veech_group_is_even(o) 
+        return false
+    end
+
+    n = 2
+    n_inv = 1 / 2
+	x = horizontal_perm(o)
+	y = vertical_perm(o)
+	g = genus(o)
+
+    L = point_reflections(o)
+	L = filter(i -> order(i) == 2, L)
+
+	if L == []
+        return false
+    end
+
+    # helper to find all elements in x that are not in y
+    function list_diff(x, y)
+        set_1 = Set(x)
+        set_2 = Set(y)
+        diff = setdiff(set_1, set_2)
+        return collect(diff)
+    end
+
+    degree_list = collect(1:degree(o))
+    for sigma in L
+        # fixpoints
+        b = 0
+        b = b + length(list_diff(degree_list, moved_points(sigma)))
+        b = b + length(list_diff(degree_list, moved_points(sigma*x)))
+        b = b + length(list_diff(degree_list, moved_points(sigma*y)))
+
+        for i in degree_list
+            if i^(sigma * x^-1 * y^-1) == i^(y*x*(x*y)^-1)
+                b = b + 1
+            end
+        end
+
+        # TODO can n_inv yields floating point errors?
+        if n_inv * (g - 1 - b * 0.5) + 1 == 0
+            return true
+        end
+    end
+
+    return false
+end
+
+function cylinder_structure(o::Origami)
+    gap_obj = GAP.Globals.CylinderStructure(GapObj(o))
+    cyl_lists = Vector{Vector{Int}}(gap_obj)
+    # TODO do we want the structure as a list of length-two lists or as
+    # a list of tuples? Current solution uses tuples.
+    cyl_tuples = [(l[1], l[2]) for l in cyl_lists]
+    return cyl_tuples
+end
+
+function veech_group_and_orbit(o::Origami)
+    return GAP.Globals.VeechGroupAndOrbit(GapObj(o))
+end
+
+function veech_group_is_even(o::Origami)
+    return GAP.Globals.VeechGroupIsEven(GapObj(o))::Bool
+end
+
+function are_equivalent(o1::Origami, o2::Origami)
+    return GAP.Globals.OrigamisEquivalent(GapObj(o1), GapObj(o2))::Bool
+end
+
+# TODO why is this not in canonical.jl?
+function normalform_conjugators(o::Origami)
+    x = horizontal_perm(o)
+    y = vertical_perm(o)
+    n = degree(o)
+    G = []
```

Comment:
This sets `G` to an empty `Vector{Any}`, i.e., is type unstable.

But I think this should work, with some more tweaks below
```suggestion
    G = PermGroupElem[]

    sym = parent(x)
```